quinn-proto/src/config.rs

```diff
@@ -973,10 +978,29 @@ impl ClientConfig {
         Self {
             transport: Default::default(),
             crypto,
+            initial_dst_cid_provider: Arc::new(|| {
+                RandomConnectionIdGenerator::new(MAX_CID_SIZE).generate_cid()
+            }),
             version: 1,
         }
     }
 
+    /// Configure how to populate the destination CID of the initial packet when attempting to
+    /// establish a new connection.
+    ///
+    /// By default, it's populated with random bytes with reasonable length, so unless you have
+    /// a good reason, you do not need to change it.
+    ///
+    /// When prefer to override the default, please note that the generated connection ID MUST be
+    /// at least 8 bytes long and unpredictable, as per section 7.2 of RFC 9000.
+    pub fn initial_dst_cid_provider<Provider: Fn() -> ConnectionId + Send + Sync + 'static>(
+        &mut self,
+        provider: Provider,
```

Comment:
Prefer to take an `Arc<dyn _>` here rather than wrapping it internally. That way if the user already has a boxed trait object, it doesn't have to be re-boxed.